java/test/jmri/util/JUnitAppender.java

```diff
@@ -323,168 +347,171 @@ private static int clearBacklog(Level level) {
 
     /**
      * Remove any messages stored up, returning how many of WARN or higher
-     * severity there are. This is used to skip over messages that don't matter,
-     * e.g. during setting up a test. Removed messages are not sent for further
-     * logging.
+     * severity there are.
+     * This is used to skip over messages that don't matter,
+     * e.g. during setting up a test.
+     * Removed messages are not sent for further logging.
      *
      * @return count of skipped messages of WARN or more specific level
      * @see #clearBacklog(Level)
      */
+    @CheckReturnValue
```

Comment:
Is this needed?  `@CheckReturnValue` is generally used when a routine has no side effects, so that if you don't check the return value there's no point in doing the call.  But this method does have side effects.

Reply:
Thanks for looking over @bobjacobsen , to save confusion I've removed these and added further comment within the JavaDoc.
What I was after is an annotation that ensures the return value is checked from a method WITH side effects to prevent it just being called like a void just to clear the backlog!